pinot-segment-local/src/main/java/org/apache/pinot/segment/local/io/compression/GzipCompressor.java

```diff
@@ -43,14 +43,17 @@ public int compress(ByteBuffer inUncompressed, ByteBuffer outCompressed)
     _compressor.setInput(inUncompressed);
     _compressor.finish();
     _compressor.deflate(outCompressed);
+    outCompressed.putInt(inUncompressed.limit()); // append uncompressed size
```

Comment:
It should be safer to use `Deflater.getBytesRead()` in case the input buffer doesn't start from position 0. Alternatively we can call `inUncompressed.remaining()` before feeding it into the `Deflater`